activejob/lib/active_job/instrumentation.rb

```diff
@@ -17,16 +17,14 @@ module Instrumentation #:nodoc:
 
     private
       def instrument(operation, payload = {}, &block)
-        enhanced_block = ->(event_payload) do
+        payload[:job] = self
+        payload[:adapter] = queue_adapter
+
+        ActiveSupport::Notifications.instrument("#{operation}.active_job", payload) do
           block.call if block
-          if defined?(@_halted_callback_hook_called) && @_halted_callback_hook_called
-            event_payload[:aborted] = true
-            @_halted_callback_hook_called = nil
-          end
+          payload[:aborted] = @_halted_callback_hook_called if defined?(@_halted_callback_hook_called)
+          @_halted_callback_hook_called = nil
         end
-
-        ActiveSupport::Notifications.instrument \
-          "#{operation}.active_job", payload.merge(adapter: queue_adapter, job: self), &enhanced_block
```

Comment:
I rewrote this code to ensure that `ActiveSupport::Notifications.instrument` receives the same `payload` that `JobRuntime#instrument` modifies.  Technically, the only change required here was `payload.merge` => `payload.merge!`, but I felt the rewritten code was more clear.  However, if this is considered too cosmetic, I can minimize the diff.